include/ck_tile/ops/epilogue/cshuffle_epilogue.hpp

```diff
@@ -1,194 +1,194 @@
+    static constexpr bool UseRawStore       = Problem::UseRawStore;
+    static constexpr index_t kBlockSize     = Problem::kBlockSize;
+    static constexpr index_t kMPerBlock     = Problem::kMPerBlock;
+    static constexpr index_t kNPerBlock     = Problem::kNPerBlock;
+    static constexpr index_t kMWave         = Problem::kMWave;
+    static constexpr index_t kNWave         = Problem::kNWave;
+    static constexpr index_t kMPerXdl       = Problem::kMPerXdl;
+    static constexpr index_t kNPerXdl       = Problem::kNPerXdl;
+    static constexpr index_t kKPerXdl       = Problem::kKPerXdl;
+    static constexpr index_t isCTransposed  = Problem::isCTransposed;
+    static constexpr index_t kMPerIteration = kMPerXdl * kMWave;
+    static constexpr index_t kNPerIteration = kNPerXdl * kNWave;
+
+    using WG = WarpGemmMfmaDispatcher<ODataType,
+                                      ODataType,
+                                      AccDataType,
+                                      kMPerXdl,
+                                      kNPerXdl,
+                                      kKPerXdl,
+                                      isCTransposed>;
+
+    using CWarpDstr   = typename WG::CWarpDstr;
+    using CWarpTensor = typename WG::CWarpTensor;
+
+    CK_TILE_HOST_DEVICE static constexpr bool IsOutputTransposed() { return true; }
+
+    /**
+     * @brief Get the vector store size for C tensor.
+     *
+     * @note The vector store size for output C tensor would depend on multiple factors
+     *       like its data layout and warp gemm C transposition. In general it would
+     *       be the number of consecutive elements in contiguous C dimension hold by
+     *       single thread.
+     *
+     * @return The vector store size for C tensor.
+     */
+    CK_TILE_HOST_DEVICE static constexpr auto GetVectorSizeC()
     {
-        const auto& current_window_origin = o_dram_window_tmp.get_window_origin();
-
-        // Compute the tile coordinates by dividing the window origin by the tile sizes
-        index_t tile_coords[CK_TILE_MAX_RANK] = {0};
-        for(index_t i = 0; i < kRank; ++i)
+        // N is contiguous dimension
+        if constexpr(std::is_same_v<CLayout, tensor_layout::gemm::RowMajor>)
         {
-            tile_coords[i] = current_window_origin[i] / tile_sizes[i];
-            // printf("The tile_coord is: %d", tile_coords[i]);
+            // In this case each thread has multiple consecutive elements in
+            // N dimension, however consecutive threads' elements have stride.
+            constexpr index_t NDimY         = CWarpDstr::NDimY;
+            constexpr auto c_warp_y_lengths = CWarpDstr{}.get_ys_to_d_descriptor().get_lengths();
+            static_assert(WG::WarpGemmAttribute::Impl::kCM1PerLane ==
+                          c_warp_y_lengths.get(number<NDimY - 1>{}));
+            return c_warp_y_lengths.get(number<NDimY - 1>{});
         }
```

Comment:
This is not true. The objective of cshuffle is maximization of vector length for output store. This code tells you just about the layout in registers after WarpGemm. And in particular for MFMA instruction it would definitely be smaller than what we could have at maximum. For MFMA 32x32x8 this would result in just 4 elements: https://github.com/ROCm/composable_kernel/blob/41bab984d637b841769368f81bdac9a8dba419f9/include/ck_tile/ops/gemm/warp/warp_gemm_attribute_mfma_impl.hpp#L92